### PR TITLE
Cr 1071 pgp

### PR DESCRIPTION
**Pull request by rpcatling**
# Motivation and Context
CR-1071 requires the title, forename and surname returned on HARD refusal details to be encrypted.
It also requires EXTRAORDINARY refusal details to not send any of the above , encrypted or otherwise.

NOTE: cucumber test needs changes to match.

# What has changed
- I have adapted the PGP encryption code existing in the FSDR codebase and copied it to Contact Centre Service.
- The decrypt code is housed in the test area, since we don't use it in production. I needed to fix the decrypt code that came from FSDR to allow testing of encryption using multiple public keys.
- Public keys are picked up on the classpath.
- The title, forename and surname fields in Refusal details are PGP encrypted, then Base64 encoded before sending to RabbitMQ
- The encrypted fields are now quite long (over 1000 characters).

# How to test?
- Unit tests show how. (see PgpEncryptTest).

### CODE REVIEW

File: src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/util/PgpEncryptTest.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  static final String PRIVATE_KEY_1 = "pgp/priv-key1.asc";
+  static final String PRIVATE_KEY_2 = "pgp/priv-key2.asc";
+
+  public static final String PASS_PHRASE = "Good Golly Miss Molly";
+  public static final String PASS_PHRASE2 = "Bless My Soul";
+
+  static final String TEST_STRING =
+      "God grant me the serenity to accept the things I cannot change, "
+          + "Courage to change the things I can, "
+          + "and Wisdom to know the difference.";
```
Comment (cookel2) [inline src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/util/PgpEncryptTest.java:29]: What if you can change something but God doesn't want you to?

**Review by cookel2 — APPROVED** [review]
Looks good but I'm not sure about what St Francis prayed : )

**Review by PeterBochel — APPROVED** [review]
Approved

File: src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/util/PgpEncryptTest.java (hunk 1 above)
Comment (rpcatling) [reply src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/util/PgpEncryptTest.java:31]: Don't know!  I inherited the test string from Martin!

**Pull request merged**